src/components/fair/booth.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from "react"
+import * as Relay from "react-relay"
+import styled from "styled-components"
+
+import Artworks from "../artwork_grid"
+import Button from "../buttons/ghost"
+import FollowButton from "../follow"
+import Text from "../text"
+
+import * as fonts from "../../assets/fonts"
+
+const PageSize = 8
+
+interface Props extends RelayProps, React.HTMLProps<FairBooth> {
+  show: any
+  relay?: any,
+  onContactGallery?: (showId: number) => any,
+}
+
+interface State {
+  loading: boolean,
+}
+
+export class FairBooth extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      loading: false,
+    }
+  }
+
+  loadMoreArtworks() {
+    const hasMore = this.props.show.artworks.pageInfo.hasNextPage
+    if (!this.state.loading && hasMore) {
+      this.setState({ loading: true }, () => {
+        this.props.relay.setVariables({
+          artworksSize: this.props.relay.variables.artworksSize + PageSize,
+        }, readyState => {
+          if (readyState.done) {
+            this.setState({ loading: false })
+          }
+        })
+      })
+    }
+  }
+
+  render() {
+    const { show, onContactGallery, relay } = this.props
+
+    const showLocation = show.location && show.location.display && (
+      <LocationText textSize="small" textStyle="secondary">
+        {show.location.display}
+      </LocationText>
+    )
+
+    const artworksLeft =  show.counts.artworks - relay.variables.artworksSize
+
+    const loadMoreButton = (
+      <LoadMoreContainer>
+        <LoadMoreButton onClick={() => this.loadMoreArtworks()}>
```

Comment:
This could be just `<LoadMoreButton onClick={this.loadMoreArtworks}>`, no need to create a new function on each render. Does require you to bind `loadMoreArtworks` to `this` in the `constructor`, though.